sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -771,6 +773,62 @@ def fill_matrix(*args):
               requires_partition_by=partitionings.Singleton(),
               proxy=proxy))
 
+  @frame_base.args_to_kwargs(pd.DataFrame)
+  @frame_base.populate_defaults(pd.DataFrame)
+  def corrwith(self, other, axis, **kwargs):
+    if axis not in (0, 'index'):
+      raise NotImplementedError('corrwith(axis=%r)' % axis)
+    if not isinstance(other, frame_base.DeferredFrame):
+      other = frame_base.DeferredFrame.wrap(
+          expressions.ConstantExpression(other))
+
+    if isinstance(other, DeferredSeries):
+      proxy = self._expr.proxy().corrwith(other._expr.proxy())
+      self, other = self.align(other, axis=0, join='inner')
+      corrs = [self[col].corr(other, **kwargs) for col in proxy.index]
+      def fill_dataframe(*args):
+        result = proxy.copy(deep=True)
+        for col, value in zip(proxy.index, args):
+          result[col] = value
+        return result
+      with expressions.allow_non_parallel_operations(True):
+        return frame_base.DeferredFrame.wrap(
+          expressions.ComputedExpression(
+            'fill_dataframe',
+            fill_dataframe,
+            [corr._expr for corr in corrs],
+            requires_partition_by=partitionings.Singleton(),
+            proxy=proxy))
+
+    elif isinstance(other, DeferredDataFrame):
+      proxy = self._expr.proxy().corrwith(other._expr.proxy())
+      self, other = self.align(other, axis=0, join='inner')
+      valid_cols = list(
+          set(self.columns)
+          .intersection(other.columns)
+          .intersection(proxy.index))
+      corrs = [self[col].corr(other[col], **kwargs) for col in valid_cols]
+      def fill_dataframe(*args):
```

Comment:
```suggestion
      # Generate expressions to compute the actual correlations
      corrs = [self[col].corr(other[col], **kwargs) for col in valid_cols]
      # Combine the results
      def fill_dataframe(*args):
```

It took me a while to realize this is what was going on, hopefully this will expedite it for future readers.